src/libraries/System.Net.Requests/src/System/Net/HttpWebResponse.cs

```diff
@@ -399,7 +399,7 @@ public override int Read(byte[] buffer, int offset, int count)
 
             public override int Read(Span<byte> buffer)
             {
-                int readBytes = innerStream.Read(buffer.Slice(0, Math.Min(buffer.Length, maxSize)));
+                int readBytes = innerStream.Read(buffer.Slice(0, (int)Math.Min(buffer.Length, maxSize)));
                 maxSize -= readBytes;
```

Comment:
Nit: I find it confusing to have primary constructor parameters be mutated... it's also maxSize to a consumer but to the implementation it's really more about how much data remains. Could we put it into a field and then use/mutate the field here? e.g.
```C#
private long _maxRemainingLength = maxSize;
```
?